src/Components/Components/src/ParameterCollectionExtensions.cs

```diff
@@ -120,8 +203,34 @@ public WritersForType(Type targetType)
                     }
 
                     WritersByName.Add(propertyName, propertySetter);
+
+                    if (parameterAttribute != null && parameterAttribute.CaptureExtraAttributes)
+                    {
+                        // This is an "Extra" parameter.
+                        //
+                        // There should only be one of these.
+                        if (CaptureExtraAttributesWriter != null)
+                        {
+                            ThrowForMultipleCaptureExtraAttributesParameters(targetType);
+                        }
+
+                        // It must be able to hold a Dictionary<string, object> since that's what we create.
+                        if (!propertyInfo.PropertyType.IsAssignableFrom(typeof(Dictionary<string, object>)))
```

Comment:
Any reason this isn't an `IDictionary<string, object>`?

Reply:
Because that's more restrictive than required.